### PR TITLE
Goesdump enhancement (Fix #11 opensatelliteproject/goesdump)

### LINKED ISSUE #11: Add a value to the goesdump configuration file which specifies the syslog facility
Link: https://github.com/opensatelliteproject/OpenSatelliteProject/issues/9

### PR DESCRIPTION
**Pull request by n2cr**
Adds 2 new keywords to the config file:

"SysLogServer" - The host name or IP address or the remote syslog server.
                            Default is "localhost".

"SysLogFacility" - The syslog facility (see <sys/syslog.h>) which should be
                             used in syslog messages.  Default is "LOG_USER".

### CODE REVIEW

File: XRIT/Log/SyslogClient.cs [hunk 1]
```diff
@@ -40,12 +41,36 @@ public class SyslogClient {
...
             if (IsActive) {
-                int priority = message.Facility * 8 + message.Level;
+                int priority = (int)FacilityMap[message.Facility] * 8 + message.Level;
                 string msg = System.String.Format("<{0}>{1} {2} {3}", priority, DateTime.Now.ToString("MMM dd HH:mm:ss"), "XRIT", message.Text);
                 byte[] bytes = System.Text.Encoding.ASCII.GetBytes(msg);
                 udpClient.Send(bytes, bytes.Length);
             }
         }
+
+        public static void InitFacilityMap() {
```
Comment (racerxdl) [inline XRIT/Log/SyslogClient.cs]: That should probably be private to avoid issues if someone calls it somwhere.

File: goesdump/HeadlessMain.cs [hunk 2]
```diff
@@ -36,6 +36,7 @@ public class HeadlessMain {
 
         private static List<ConsoleMessage> messageList = new List<ConsoleMessage>();
         private static Mutex messageListMutex = new Mutex();
+        private SyslogClient syslog = new SyslogClient();
```
Comment (racerxdl) [inline goesdump/HeadlessMain.cs]: You don't need to create it here, since the instance on headless main is just for testing if the syslog is configured.

**Review by racerxdl — APPROVED** [review]

File: XRIT/Log/SyslogClient.cs (hunk 1 above)
Comment (n2cr) [reply XRIT/Log/SyslogClient.cs]: Agreed.  Will fix.

**Comment by racerxdl** [comment]
That will also close https://github.com/opensatelliteproject/OpenSatelliteProject/issues/9

**Pull request merged**